Gemfile

```diff
@@ -56,3 +57,12 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+#simple_form and devise
+gem 'simple_form'
+gem 'devise'
+
+#Testing gems
+gem 'rails-controller-testing'
```

Comment:
Cool - that does make sense. So these are put in the Gemfile and just get used when in the appropriate environment?

Reply:
That's right. You only want `rails-controller-testing` in your test environment, so it would go in a `group :test do ...` block. Things generally go in the `:development, :test` block if used for tests but also have a command line element/generator (e.g. RSpec).

Anything which your app actually needs to run should be outside a block.